Update CSSPseudoElement docs, and add Element.pseudo() (#44275)
mdn/content: files/en-us/web/api/csspseudoelement/pseudo/index.md

---
title: "CSSPseudoElement: pseudo() method"
short-title: pseudo()
slug: Web/API/CSSPseudoElement/pseudo
page-type: web-api-instance-method
browser-compat: api.CSSPseudoElement.pseudo
---

{{APIRef("CSSOM view API")}}

The **`pseudo()`** method of the {{domxref("CSSPseudoElement")}}

> interface returns a `CSSPseudoElement` object representing the [CSS](/en-US/docs/Web/CSS) [pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements) of the specified type associated with the parent pseudo-element.

Reviewer: ```suggestion
The **`pseudo()`** method of the {{domxref("CSSPseudoElement")}} interface returns a `CSSPseudoElement` object representing the [CSS](/en-US/docs/Web/CSS) [pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements) of the specified type associated with the parent pseudo-element.
```

> interface returns a `CSSPseudoElement` object representing the [CSS](/en-US/docs/Web/CSS) [pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements) of the specified type associated with the parent pseudo-element.

Reviewer: This reads like "a pseudo element blah pseudo element blah blah pseudo-element." I appreciate that this is what it does, but can we make it less repetetive. Perhaps punctuate a bit more. I think it is a bit confusing also because again, it isn't clear how a pseudo can have a pseudo element.

Note sure exactly how, but something like ...

```suggestion
The **`pseudo()`** method of the {{domxref("CSSPseudoElement")}} interface returns a `CSSPseudoElement` instance representing a specific nested [pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements).

It is used to target a pseudo-element that is attached to another pseudo-element, rather than directly to a standard DOM element. For example, if a `::before` pseudo-element generates a list marker, this method can retrieve the `::marker` nested inside that `::before`. You call the method on the parent pseudo-element and pass the type of the nested child pseudo-element as an argument.
```

Author: Sure, that's pretty good. I've mostly used your suggestion and tweaked it a bit.

This is useful for returning a representation of a [nested pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements#nesting_pseudo-elements), for example the `::marker` pseudo-element in `::after::marker`.

## Syntax

```js-nolint
pseudo(type)
```

### Parameters

> - `type`

Reviewer: IF this can't be any type, and I don't think it can be, you need to list the allowed types. Either here or in the type property and link

Author: Yeah, I don't know why I did this. I think I read something somewhere that said the list of supported types was different in the spec versus Chrome, and I thought I'd handle the supported types in BCD. But the spec actually lists the valid types as `::after`, `::before`, and `::marker`.

I've updated the information here and on `Element.pseudo` and `CSSPseudoElement.type` to say that only those three are supported.

```html live-sample___basic
<p>New York's hottest club is...</p>
<output></output>
```

#### CSS

We give the `<p>` element's {{cssxref("::after")}} pseudo-element some {{cssxref("content")}} and set its {{cssxref("display")}} to `list-item` so it will generate a `::marker`. We also apply some basic styles.

```css hidden live-sample___basic
body {
  width: 80%;
  margin: 0 auto;
}
```

```css live-sample___basic
p {
  background-color: violet;
  padding: 20px;
}

p::after {
  content: "Crease";
  background-color: cadetblue;
  padding: 20px;
  display: list-item;
}

p::after::marker {
  content: "🔹";
}
```

#### JavaScript

In our script, we grab references to our `<p>` and `<output>` elements, and retrieve `CSSPseudoElement` objects via the `pseudo()` method representing the `<p>` element's `::after` pseudo-element, and the `::after` pseudo-element's `::marker` pseudo-element. We then log some details of the child pseudo-element to our `<output>` element.

```js live-sample___basic
const pElem = document.querySelector("p");
const output = document.querySelector("output");
const pseudoElem = pElem.pseudo("::after");
const pseudoPseudoElem = pseudoElem.pseudo("::marker");

> output.textContent = `${pseudoPseudoElem.type} pseudo-element. Parent: ${pseudoPseudoElem.parent}. Originating parent element: ${pseudoPseudoElem.element}`;

Reviewer: As before, maybe in a try...catch

Author: Done

browser-compat: api.CSSPseudoElement.pseudo
---

{{APIRef}}

The **`pseudo()`** method of the {{domxref("CSSPseudoElement")}} interface returns a `CSSPseudoElement` instance representing a specific [nested pseudo-element](/en-US/docs/Web/CSS/Reference/Selectors/Pseudo-elements#nesting_pseudo-elements).

## Syntax

```js-nolint
pseudo(type)
```

### Parameters

- `type`
  - : A string representing the type of pseudo-element to return a representation of. Valid values are:
    - {{cssxref("::after")}}
    - {{cssxref("::before")}}
    - {{cssxref("::marker")}}

### Return value

A {{domxref("CSSPseudoElement")}} object instance, or `null` if `type` is not equal to a valid pseudo-element type.

## Description

The `CSSPseudoElement.pseudo()` method is used to target a pseudo-element that is attached to another pseudo-element, rather than directly to a standard DOM element. For example, if a `::before` pseudo-element generates a list marker — selectable via `::before::marker` — this method can retrieve the `::marker` nested inside that `::before`. You call the method on the parent pseudo-element and pass the type of the nested child pseudo-element as an argument.

> Provided its `type` property contains a valid pseudo-element type, `pseudo()` will always return a `CSSPseudoElement` instance, even if that pseudo-element hasn't been generated on the calling pseudo-element.

Reviewer: ```suggestion
Provided its `type` parameter contains a valid pseudo-element type, `pseudo()` will always return a `CSSPseudoElement` instance, even if that pseudo-element hasn't been generated on the calling pseudo-element.
```

Author: I fixed this one myself, after seeing the other instance.
